src/sonic-yang-models/yang-models/sonic-snmp.yang

```diff
@@ -70,5 +70,89 @@ module sonic-snmp {
         }
       }
     }
+    container SNMP_USER {
+      list SNMP_USER_LIST {
+        key name;
+        description
+          "List of users.";
+
+        leaf name {
+          mandatory true;
+          type string {
+              length "4..32";
+              pattern '[^ @,\\' +"']*" {
+                  error-message 'Invalid snmp user string (Valid chars are ASCII printable except SPACE, single quote,"@", "," and "\")';
+              }
+          }
+          description
+            "Index into the user list which must be the user name.";
+        }
+
+        leaf SNMP_USER_TYPE {
+          mandatory true;
+          type enumeration {
+            enum noAuthNoPriv;
+            enum AuthNoPriv;
+            enum Priv;
+          }
+          description
+            "Authentication and encryption method used for the user, can be noAuthNoPriv,AuthNoPriv or Priv";
+        }
+        leaf SNMP_USER_PERMISSION {
+          mandatory true;
+          type enumeration {
+            enum RO;
+            enum RW;
+          }
+          description
+            "Specify if the user permission can be Read only or Read write user, can be RO or RW";
+        }
+        leaf SNMP_USER_AUTH_TYPE {
+          when "current()/../SNMP_USER_TYPE != 'noAuthNoPriv'";
```

Comment:
if the `SNMP_USER_AUTH_TYPE` is `noAuthNoPriv` this configure is not required, is that correct? The leaf has mandatory = true so what should be the value the user should specify?

Reply:
In case of CLI, if the SNMP_USER_AUTH_TYPE is noAuthNoPriv, user need not specifiy any other information other than USER_NAME and PERMISSION.
So this field is mandatory only if SNMP_USER_AUTH_TYPE is NOT  noAuthNoPriv

Should we put a default value of "" empty string?
CLI implementation handles adding default value to configDB.
```
admin@1:~$ sudo config snmp user add test noAuthNoPriv RO

127.0.0.1:6379[4]> HGETALL "SNMP_USER|test"
 1) "SNMP_USER_AUTH_PASSWORD"
 2) ""
 3) "SNMP_USER_AUTH_TYPE"
 4) ""
 5) "SNMP_USER_ENCRYPTION_PASSWORD"
 6) ""
 7) "SNMP_USER_ENCRYPTION_TYPE"
 8) ""
 9) "SNMP_USER_PERMISSION"
10) "RO"
11) "SNMP_USER_TYPE"
12) "noAuthNoPriv"
```

Reply:
Need to find a good way to make it mandatory based on SNMP_USER_AUTH_TYPE  or set to a default value of empty string without complying to the constraint of minimum password length or enum.